### DISCUSSION TITLE
func1 succeeds, func2 fails, will func1 be rollback?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#1996] [Q&A] [answered]
```
void test() async {
  await transaction()async{
    await transaction()async{
      func1();
    };
    await transaction()async{
      func2();
    };
  }
}
```

**Comment by simolus3** [comment, marked as answer]
Yes exactly. When `func2()` throws, the second inner transaction will be rolled back. Then, the exception thrown by `func2` will be rethrown and, since there is no try/catch around the second transaction block, the outer transaction throws an exception too. Drift will then revert the entire transaction, which includes the changes made by `func1`.
